server/routes/api/shares/schema.ts

```diff
@@ -0,0 +1,97 @@
+import { isEmpty } from "lodash";
+import isUUID from "validator/lib/isUUID";
+import { z } from "zod";
+import { SHARE_URL_SLUG_REGEX, SLUG_URL_REGEX } from "@shared/utils/urlHelpers";
+import BaseSchema from "../BaseSchema";
+
+export const SharesInfoSchema = BaseSchema.extend({
+  body: z
+    .object({
+      id: z.string().uuid().optional(),
+      documentId: z
+        .string()
+        .optional()
+        .refine(
+          (val) => (val ? isUUID(val) || SLUG_URL_REGEX.test(val) : true),
+          {
+            message: "must be uuid or url slug",
+          }
+        ),
+    })
+    .refine((body) => !(isEmpty(body.id) && isEmpty(body.documentId)), {
+      message: "id or documentId is required",
+    }),
+});
+
+export type SharesInfoReq = z.infer<typeof SharesInfoSchema>;
+
+export const SharesListSchema = BaseSchema.extend({
+  body: z.object({
+    sort: z
+      .string()
+      .refine(
+        (val) =>
```

Comment:
Can we use the same method as assertSort rather than hard coding? 
https://github.com/outline/outline/blob/8e4270c3219097f7406a555e1cf5d56fabcafd4f/server/validation.ts#L56

Reply:
My thinking was to only include those attributes which seem relevant for sorting, rather than including all. It's a decent tradeoff though :+1: